dask_jobqueue/core.py

```diff
@@ -138,6 +139,7 @@ class JobQueueCluster(Cluster):
     cancel_command = None
     scheduler_name = ''
     _adaptive_options = {'worker_key': lambda ws: _job_id_from_worker_name(ws.name)}
+    job_id_regexp = r'\d+'
```

Comment:
Put the full regexp here, i.e. `(?P<job_id>\d+)`. This way this is fully customisable in derived class. Currently you are assuming that the regexp starts with `(?<job_id>` in `_job_id_from_submit_output` which may not be appropriate.

Reply:
Done.